Src/Plugins/Tests.Security/EndToEndTests.cs

```diff
@@ -138,5 +139,43 @@ protected void RunAllTests()
 
             RunTest(inputFiles, expectedOutputResourceMap, enforceNotificationsFree: true);
         }
+
+        protected void ValidateOutputFiles()
+        {
+            Directory.Exists(ProductTestDataDirectory).Should().BeTrue();
+
+            string testsDirectory = Path.Combine(ProductTestDataDirectory, @"ExpectedOutputs\");
+
+            var sb = new StringBuilder();
+            foreach (string testFile in Directory.GetFiles(testsDirectory))
+            {
+                string fileName = Path.GetFileName(testFile);
+                var sarifLog = SarifLog.Load(testFile);
+                if (sarifLog == null)
+                {
+                    sb.AppendLine($"SarifLog from file '{fileName}' should not be null.");
+                    continue;
+                }
+
+                if (sarifLog.Runs[0].Results == null || sarifLog.Runs[0].Results.Count == 0)
+                {
+                    sb.AppendLine($"Results from file '{fileName}' should not be empty.");
+                    continue;
+                }
+
+                //foreach (Result result in sarifLog.Runs[0].Results)
+                //{
+                //    var fingerprint = new Fingerprint(result.Fingerprints[SearchSkimmer.ValidationFingerprintV2]);
+                //    if (!result.Locations[0].PhysicalLocation.Region.Snippet.Text.StartsWith(fingerprint.Secret))
```

Comment:
>fingerprint.Secret [](http://example.com/codeflow?start=95&length=18)

I'm still trying to figure out how to test a requirement where the secret must appear in the region.

There is one scenario that this test does not work:
1. we receive a base64 string -> we make the conversion
2. when we have a rule that brings the entire context to be analyzed in our validator

Reply:
Why isn't this invariant in the actual code somewhere? Verifying this in functional tests provides limited value.

Honestly, what we should probably do is stop requiring plug-ins to populate the region override. In cases when a rule resurns a fingerprint, if 'Secret' is populated, we should consider having the tool itself set the region to that data.

The scanner itself will know if it is base64-decoding or not and could skip this step.

Reply:
btw - even if we decide to have 'test only' validation, instead of building it into a post-processing step, you could add conditionally executing code deep in the scanner itself. 

Sometimes I create a static property 'TestingInProgress' that I use to conditionally control additional verifications.

Reply:
changed the logic in searchskimmer to automatically use secret match if that exists.

But, if any validator still needs to overwrite, it would still work.